### ISSUE TITLE
Order of arrays for dkim_options()

### ISSUE THREAD

**Issue by countsudoku** [trusteddomainproject/OpenDKIM#179]
The documentation for [`dkim_options()`](https://github.com/trusteddomainproject/OpenDKIM/blob/5c539587561785a66c1f67f720f2fb741f320785/libopendkim/docs/dkim_options.html) states for some array options: they should be __unordered__ (`DKIM_OPTS_OVERSIGNHDRS`, `DKIM_OPTS_SIGNHDRS` and `DKIM_OPTS_SKIPHDRS`) and for others they should be __ordered__ (`DKIM_OPTS_MUSTBESIGNED`, and `DKIM_OPTS_REQUIREDHDRS`). But the documentation does not tell us, in which order the ordered items should be. Also I did not find any reason by digging through the code, why some of those array options must have a specific order (maybe except of some performance benefit, if some header is found first, but this might depend on the input mail).

So can somebody elaborate, why some of these options should be __ordered__? Otherwise I suggest the __ordered__, __unordered__ stuff should be removed from the documentation.

**Comment by thegushi** [comment]
You're correct — both arrays are iterated linearly in the code with no ordering dependency. Fixed in PR #304.
